pkg/logs/decoder/line_handler.go

```diff
@@ -79,43 +79,42 @@ func (h *SingleLineHandler) run() {
 // the limit and that the length of the line is properly tracked
 // so that the agent restarts tailing from the right place.
 func (h *SingleLineHandler) process(line []byte) {
-	lineLen := len(line)
-	line = bytes.TrimSpace(line)
-	if len(line) == 0 {
+	isTruncated := h.shouldTruncate
+	h.shouldTruncate = false
+
+	rawLen := len(line)
+	if rawLen < h.lineLimit {
+		// lines are delimited on '\n' character when bellow the limit
+		// so we need to make sure it's properly accounted
+		rawLen++
+	}
+
+	content, status, timestamp, err := h.parser.Parse(line)
```

Comment:
Most of the time, logs are prepend with a header which contains the severity and a timestamp so if we add the truncated flag at the beginning of a content, it could break the parsing logic afterward. I also feel like that we want to truncate the content and not the frame.